crates/common/tedge_config/src/tedge_config_cli/models/mod.rs

```diff
@@ -25,3 +26,35 @@ pub use self::ipaddress::*;
 pub use self::port::*;
 pub use self::seconds::*;
 pub use self::templates_set::*;
+pub use tedge_utils::timestamp;
+
+#[derive(
+    Debug, Display, Clone, Eq, PartialEq, doku::Document, serde::Serialize, serde::Deserialize,
+)]
+#[strum(serialize_all = "camelCase")]
+pub enum AutoLogUpload {
+    Never,
+    Always,
+    OnFailure,
+}
+
+#[derive(thiserror::Error, Debug)]
+#[error("Failed to parse flag: {input}. Supported values are: 'never', 'always' or 'onFailure'")]
+pub struct InvalidAutoLogUploadConfig {
+    input: String,
+}
+
+impl FromStr for AutoLogUpload {
+    type Err = InvalidAutoLogUploadConfig;
+
+    fn from_str(input: &str) -> Result<Self, Self::Err> {
+        match input {
+            "never" => Ok(AutoLogUpload::Never),
+            "always" => Ok(AutoLogUpload::Always),
+            "onFailure" => Ok(AutoLogUpload::OnFailure),
+            _ => Err(InvalidAutoLogUploadConfig {
+                input: input.to_string(),
+            }),
+        }
+    }
+}
```

Comment:
I think it's better to create a new file under `tegge_config/cli/models/` and move the code there. All other enum or struct that are used in tedge config are stored in that directory.

Reply:
Yeah, I noticed that pattern being followed for other config enums like `apt_conifg.rs` and `c8y_software_management.rs`. If that's an accepted convention and not mere accident, I can move this one also to a separate module. But, to be honest, a module per enum/struct looked like an overkill to me, which is why I consciously kept this one at the top module level itself. I usually create dedicated modules when there are multiple related structs/enums that are to be kept together.

Reply:
I understand your point, but having only one enum in the top level and all others in the separated files doesn't look well organized. All enums for config should be treated in some clear rule, either all in the top or all in the separated files, or some light-weight enums in the top and heavy-weight in the separated files. We need a clear rule.

Reply:
> some light-weight enums in the top and heavy-weight in the separated files. We need a clear rule.

That's what I'd follow too. The other lightweight enums can also be moved into the top level in a follow-up PR.